core/src/main/java/net/opentsdb/query/ReadCacheQueryPipelineContext.java

```diff
@@ -337,17 +339,24 @@ public Void call(ArrayList<Void> arg) throws Exception {
             }
           }
         }
-        
-        final Set<String> serdes_sources = computeSerializationSources();
-        for (final String source : serdes_sources) {
-          if (serdes_filter != null && !serdes_filter.isEmpty()) {
-            if (serdes_filter.contains(source)) {
-              countdowns.put(source, new AtomicInteger(sinks.size()));  
-            }
-          } else {
-            countdowns.put(source, new AtomicInteger(sinks.size()));
-          }
+// TODO --------------------------*(*        
+        CacheQueryPlanner planner = new CacheQueryPlanner(
+            ReadCacheQueryPipelineContext.this, (QueryNode) ReadCacheQueryPipelineContext.this);
```

Comment:
Why are you passing the same object twice as the argument?

Reply:
In this case because the parent implements both. In some cases it may not.